File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Networking/Request/GetUnregisteredRacesRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ubb.BikeContest.Networking.Request
{
    [Serializable]
    public class GetUnregisteredRacesRequest : IRequest
    {
        private readonly long participantId;
        private readonly int engineCapacity;

        public GetUnregisteredRacesRequest(long participantId, int engineCapacity)
        {
            this.participantId = participantId;
            this.engineCapacity = engineCapacity;
        }

        public virtual long ParticipantId { get { return participantId; } }

        public virtual int EngineCapacity { get {  return engineCapacity; } }
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/Java/networking/src/main/java/com/ubb/protocol/request/GetUnregisteredRacesRequest.java
package com.ubb.protocol.request;

public class GetUnregisteredRacesRequest implements Request {
    private final Long participantID;
    private final Integer engineCapacity;

    public GetUnregisteredRacesRequest(Long participantID, Integer engineCapacity) {
        this.participantID = participantID;
        this.engineCapacity = engineCapacity;
    }

    public Long getParticipantID() {
        return participantID;
    }

    public Integer getEngineCapacity() {
        return engineCapacity;
    }
}
